src/evict/evict_page.c

```diff
@@ -675,9 +676,37 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
+    /* Acquire a snapshot if coming through eviction thread route. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL) && !F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      !F_ISSET(conn,
+        WT_CONN_CLOSING | WT_CONN_RECOVERING | WT_CONN_IN_MEMORY | WT_CONN_CLOSING_TIMESTAMP) &&
+      !WT_IS_HS(S2BT(session))) {
+        /*
+         * We don't have a running transaction with eviction threads. Starting a new transaction
+         * should implicitly take a snapshot as well.
+         */
+        WT_RET(__wt_txn_begin(session, txn_cfg));
+        /* Eviction threads do not need to pin anything. */
+        WT_SESSION_TXN_SHARED(session)->pinned_id = WT_TXN_NONE;
+        txn_started = true;
+    } else {
+        /*
+         * Only set this flag for application threads or whenever we are unable to start a
+         * transaction.
+         */
+        LF_SET(WT_REC_VISIBLE_ALL);
+    }
+    WT_ASSERT(session, !txn_started || F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT));
+
     /* Reconcile the page. */
     ret = __wt_reconcile(session, ref, NULL, flags);
 
+    /* End the transaction if we started one. */
+    if (txn_started) {
+        WT_SESSION_TXN_SHARED(session)->pinned_id = WT_TXN_NONE;
```

Comment:
What is the reason behind setting the `pinned_id` to NONE here again? If there is any reason, can you add a comment?

Reply:
I have fixed this. I added setting `pinned_id` to `WT_TXN_NONE` when starting a transaction but didn't remove it from other places. Thank you for pointing it out.